src/sql/base/browser/ui/radioButton/radioButton.ts

```diff
@@ -74,7 +74,7 @@ export class RadioButton extends Widget {
 	}
 
 	public set checked(val: boolean) {
-		if (this.inputElement.checked !== this._internalCheckedStateTracker) {
+		if (val !== this._internalCheckedStateTracker) {
```

Comment:
Hmm, I feel like we should be doing two things here : 

1. Check if `val !== this._internalCheckedStateTracker`. If that's true then update the internal state tracker and fire the event
2. Check if `this.inputElement.checked !== this._internalCheckedStateTracker)` If that's true then set inputElement.checked to val (which due to 1 we know is the same as the internal state tracker)

This way we'll ensure that any time we set checked that everything is synced up correctly - right now with your change if for some reason the internal state is different than the actual checked value (which would happen if something changed the value directly through code).

Reply:
@Charles-Gagnon please hold off. This is still broken

Reply:
@Charles-Gagnon, I have fixed this scenario. Please review my PR again.